### PR TITLE
Simplify VFS initialization

### PR DESCRIPTION
**Pull request by sosthene-nitrokey**
Can be improved further once https://github.com/trussed-dev/littlefs2/pull/87 is merged.

### CODE REVIEW

File: components/boards/src/store.rs [hunk 1]
```diff
@@ -194,7 +195,9 @@ pub fn init_store<B: Board>(
         .compare_exchange_weak(false, true, Ordering::AcqRel, Ordering::Acquire)
         .expect("multiple instances of RunnerStore are not allowed");
 
-    static mut VOLATILE_STORAGE: Option<VolatileStorage> = None;
+    static mut VOLATILE_STORAGE: VolatileStorage = VolatileStorage {
+        buf: [0; 8192],
+    };
```
Comment (robin-nitrokey) [inline components/boards/src/store.rs:200]: Why not `VolatileStorage::new()`?

**Review by robin-nitrokey — CHANGES_REQUESTED** [review]

**Review by robin-nitrokey — APPROVED** [review]

**Comment by robin-nitrokey** [comment]
Note sure what causes the binary size change.  I cannot reproduce it locally.

**Comment by sosthene-nitrokey** [comment]
I do not understand the size increase. On the opposite I would have expected a size decrease.

**Comment by robin-nitrokey** [comment]
My first guess was that maybe the build environment changed and caused the size change, but running the CI for a no-op PR like https://github.com/Nitrokey/nitrokey-3-firmware/pull/240 does not cause a size difference.

**Comment by sosthene-nitrokey** [comment]
I can reproduce the size change locally.  My first thought was a missing rebase but it's not that.

**Comment by robin-nitrokey** [comment]
I only checked the output of `make bloat` locally, not of the binary size.  It is quite the coincidence that the volatile storage size is 8192 and the size change is e. g. 8176.  It looks like this is just the size of the initializer?  Using a `MaybeUninit` to defer initialization to runtime could help.

**Comment by sosthene-nitrokey** [comment]
Yes it looks like it was that. It's including the full zeros of the static? That's not good. As a result using `MaybeUninit` improves the binary size by the couple of relevant bytes.

**Review by robin-nitrokey — APPROVED** [review]

**Review by daringer — APPROVED** [review]

**Comment by daringer** [comment]
mmmh, falsely approved although the build fails - didn't check it yet

**Pull request merged**